packages/server/src/tests/utilities/TestConfiguration.ts

```diff
@@ -188,30 +189,38 @@ class TestConfiguration {
     }
   }
 
-  // MODES
-  setMultiTenancy = (value: boolean) => {
-    env._set("MULTI_TENANCY", value)
-    coreEnv._set("MULTI_TENANCY", value)
+  async withEnv(newEnvVars: Partial<typeof env>, f: () => Promise<void>) {
+    let cleanup = this.setEnv(newEnvVars)
+    try {
+      await f()
+    } finally {
+      cleanup()
+    }
   }
 
-  setSelfHosted = (value: boolean) => {
-    env._set("SELF_HOSTED", value)
-    coreEnv._set("SELF_HOSTED", value)
-  }
+  /*
+   * Sets the environment variables to the given values and returns a function
+   * that can be called to reset the environment variables to their original values.
+   */
+  setEnv(newEnvVars: Partial<typeof env>): () => void {
```

Comment:
Merged the various functions in this file that modified the environment into this one function that gives the option of cleaning up the changes.

It did highlight a bit of a problem in our `env` type: `SELF_HOSTED` is a `string | undefined` but in reality we set it to `0` when it's false, and there's code that seems to rely on this fact (by checking `if (!env.SELF_HOSTED)`. Not sure what's best to do about this. It makes using this function a bit awkward because `setEnv({ SELF_HOSTED: 0 })` is a type error.